File: src/lava/magma/runtime/_c_message_infrastructure/test/test_fastdds_from_ros.cc
// Copyright (C) 2022 Intel Corporation
// SPDX-License-Identifier: BSD-3-Clause
// See: https://spdx.org/licenses/

#include <channel/dds/fast_dds.h>
#include <core/channel_factory.h>
#include <channel/dds/dds_channel.h>

using namespace message_infrastructure;  // NOLINT

#define LOOP_NUM 100

int main() {
  auto dds_channel = GetChannelFactory()
    .GetDDSChannel("test_channel_src",
                   "test_channel_dst",
                   "rt/dds_topic",
                   10,
                   DEFAULT_NBYTES,
                   DDSTransportType::DDSUDPv4,
                   DDSBackendType::FASTDDSBackend);
  auto dds_recv = dds_channel->GetRecvPort();
  int loop = LOOP_NUM;

  dds_recv->Start();
  while (loop--) {
    MetaDataPtr res = dds_recv->Recv();
    printf("DDS recv : '%d'\n", *reinterpret_cast<char*>(res->mdata));
  }
  dds_recv->Join();
  return 0;
}

File: src/lava/magma/runtime/_c_message_infrastructure/examples/ros2/test_fastdds_from_ros.py
# Copyright (C) 2022 Intel Corporation
# SPDX-License-Identifier: BSD-3-Clause
# See: https://spdx.org/licenses/


from lava.magma.runtime.message_infrastructure import (
    ChannelQueueSize,
    GetDDSChannel,
    DDSTransportType,
    DDSBackendType
)


def test_ddschannel():
    name = 'rt/dds_topic'

    dds_channel = GetDDSChannel(
        name,
        DDSTransportType.DDSUDPv4,
        DDSBackendType.FASTDDSBackend,
        ChannelQueueSize)

    recv_port = dds_channel.dst_port
    recv_port.start()
    for _ in range(100):
        res = recv_port.recv()
        print(res)
    recv_port.join()


if __name__ == "__main__":
    test_ddschannel()
